src/Stratis.Bitcoin.Features.RPC/RPCClient.cs

```diff
@@ -1139,16 +1139,18 @@ public Task<Transaction> DecodeRawTransactionAsync(byte[] raw)
         /// <summary>
         /// getrawtransaction only returns on txn which are not entirely spent unless you run bitcoinq with txindex=1.
         /// </summary>
-        /// <param name="txid"></param>
-        /// <returns></returns>
-        public Transaction GetRawTransaction(uint256 txid, bool throwIfNotFound = true)
+        /// <param name="txid">The transaction id to retrieve.</param>
+        /// <param name="blockHash">The optional block hash in which to look.</param>
+        /// <param name="throwIfNotFound">Whether to throw an exception if not found.</param>
+        /// <returns>The transaction </returns>
+        public Transaction GetRawTransaction(uint256 txid, uint256 blockHash = null, bool throwIfNotFound = true)
```

Comment:
You might want to consider putting blockHash after throwIfNotFound. Seems like there is lots of code GetRawTransaction(txId, null, false) which could be left GetRawTransaction(txId, false) then

Reply:
Is the `throwIfNotFound` even needed? whats it's purpose?

Reply:
ah sorry this is the client ignore me!

Reply:
@mikedennis yeah indeed but I prefer having the parameters at the beginning and the weird behavioral flags at the end. Looks more tidy.